pkg/kubelet/volumemanager/volume_manager.go

```diff
@@ -361,27 +363,35 @@ func (vm *volumeManager) WaitForAttachAndMount(pod *v1.Pod) error {
 		vm.verifyVolumesMountedFunc(uniquePodName, expectedVolumes))
 
 	if err != nil {
-		// Timeout expired
 		unmountedVolumes :=
 			vm.getUnmountedVolumes(uniquePodName, expectedVolumes)
 		// Also get unattached volumes for error message
 		unattachedVolumes :=
 			vm.getUnattachedVolumes(expectedVolumes)
 
 		if len(unmountedVolumes) == 0 {
-			return nil
+			return false, nil
 		}
 
-		return fmt.Errorf(
-			"timeout expired waiting for volumes to attach or mount for pod %q/%q. list of unmounted volumes=%v. list of unattached volumes=%v",
+		if err == wait.ErrWaitTimeout {
+			return false, fmt.Errorf(
+				"timeout expired waiting for volumes to attach or mount for pod %q/%q. list of unmounted volumes=%v. list of unattached volumes=%v",
+				pod.Namespace,
+				pod.Name,
+				unmountedVolumes,
+				unattachedVolumes)
+		}
+		return true, fmt.Errorf(
+			"error waiting for volumes to attach or mount for pod %q/%q. list of unmounted volumes=%v. list of unattached volumes=%v. error: %v",
 			pod.Namespace,
 			pod.Name,
 			unmountedVolumes,
-			unattachedVolumes)
+			unattachedVolumes,
+			err)
 	}
 
 	klog.V(3).Infof("All volumes are attached and mounted for pod %q", format.Pod(pod))
-	return nil
+	return false, nil
 }
```

Comment:
Seems this func always returns false, so I wonder why we use it as a returned value?